pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -2177,6 +2178,28 @@ func (og *operationGenerator) nodeExpandVolume(
 	return true, nil
 }
 
+func (og *operationGenerator) checkForRecoveryFromExpansion(pvc *v1.PersistentVolumeClaim, volumeToMount VolumeToMount) bool {
+	resizeStatus := pvc.Status.ResizeStatus
+	allocatedResource := pvc.Status.AllocatedResources
+	featureGateStatus := utilfeature.DefaultFeatureGate.Enabled(features.RecoverVolumeExpansionFailure)
+
+	if !featureGateStatus {
+		return false
+	}
+
+	// Even though RecoverVolumeExpansionFailure feature gate is enabled, it appears that we are running with older version
+	// of resize controller, which will not populate allocatedResource and resizeStatus. This can happen because of version skew
+	// and hence we are going to keep expanding using older logic.
+	if resizeStatus == nil && allocatedResource == nil {
+		simpleMsg, detailedMsg := volumeToMount.GenerateMsg("MountVolume.NodeExpandVolume used with", "older resize controller")
```

Comment:
Is it the resize-controller in KCM or external-resizer sidecar? IMO the message should clearly tell that KCM or the resizer must be updated.